### ISSUE TITLE
Modal dialog comments

### ISSUE THREAD

**Issue by KittyGiraudel** [frend/frend.co#63]
Hey there. I am the author of [a11y-dialog](https://github.com/edenspiekermann/a11y-dialog), so allow me to give a few comments on [your current implementation of the modal component](https://github.com/frend/frend.co/tree/component/fr-dialogmodal), as [you asked on Twitter](https://twitter.com/ffffrend/status/727439890095443968).
- Default `type` attribute for the `button` element is `submit`, so make sure to change it to `button` when used for a dialog.
- The dialog itself should have the `aria-hidden` attribute set to `true` when not displayed. When displayed, the attribute can be removed, and the main content should have `aria-hidden` set to `true`. Which means the dialog should not live inside the main container. _Edit: you did it through JS, interesting._
- The dialog itself should have the `role` attribute set to `dialog`. _Edit: you did it through JS, interesting._
- The dialog first and only direct child should have the `role` attribute set to `document`. This makes it so NVDA automatically switches into document reading mode inside of the modal. NVDA had previously let you toggle the reading mode, but since many modals contain items that require document browsing mode, it matters.
- Then you need an overlay, and a way to trap the focus inside the dialog while it’s open. Feel free to check how we did it on a11y-dialog. :)

Best.

**Comment by thomasdigby** [comment]
Thanks for this @HugoGiraudel , we appreciate your input!

**Comment by thomasdigby** [comment]
Again thanks for the feedback Hugo, a few points:
- We're setting the `aria` and `role` attributes through JS as they only seem necessary when running the component with JS. We're attempting to support basic functionality for all of our components when JS is unavailable, so in the case of the modal, we've come to the conclusion it should just render as another section of the page.
- With that in mind, will a `role="document"` affect the page/accessibility tree when running without JS?
- We're trapping focus within the modal correctly although we've ran into issues when getting the `document.activeElement` on `window` `keydown`, it always seems to be returning the previous `activeElement` and not the current one. We've implemented a fairly tedious workaround in the meantime but we'd like to get closer to your work on [a11y-dialog](https://github.com/edenspiekermann/a11y-dialog/blob/master/a11y-dialog.js#L19L30). Is there anything we're missing here?

Thanks

**Comment by KittyGiraudel** [comment]
I do not remember encountering any specific issue on that matter. I’ll check if I ever have a chance in the next days (which is unlikely :().

**Comment by thomasdigby** [comment]
Merged into main branch with Hugo's suggestions.
